src/BigBlueButton.php

```diff
@@ -548,88 +404,101 @@ public function buildUrl($method = '', $params = '', $append = true)
-                throw new \RuntimeException('Unhandled curl error: ' . curl_error($ch));
-            }
-            $httpcode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
-            if ($httpcode < 200 || $httpcode >= 300) {
-                throw new BadResponseException('Bad response, HTTP code: ' . $httpcode);
-            }
-            curl_close($ch);
-            unset($ch);
+        if (!$ch) {  // @phpstan-ignore-line
+            throw new \RuntimeException('Unhandled curl error: ' . curl_error($ch));
+        }
 
-            $cookies = file_get_contents($cookiefilepath);
-            if (false !== mb_strpos($cookies, 'JSESSIONID')) {
-                preg_match('/(?:JSESSIONID\s*)(?<JSESSIONID>.*)/', $cookies, $output_array);
-                $this->setJSessionId($output_array['JSESSIONID']);
+        // JSESSIONID
+        if ($cookieFile) {
+            $cookieFilePath = stream_get_meta_data($cookieFile)['uri'];
+            $cookies        = file_get_contents($cookieFilePath);
+
+            curl_setopt($ch, CURLOPT_COOKIEFILE, $cookieFilePath);
+            curl_setopt($ch, CURLOPT_COOKIEJAR, $cookieFilePath);
+
+            if ($cookies) {
+                if (false !== mb_strpos($cookies, 'JSESSIONID')) {
+                    preg_match('/(?:JSESSIONID\s*)(?<JSESSIONID>.*)/', $cookies, $output_array);
+                    $this->setJSessionId($output_array['JSESSIONID']);
+                }
             }
+        }
 
-            return $data;
+        // PAYLOAD
+        if (!empty($payload)) {
+            curl_setopt($ch, CURLOPT_HEADER, 0);
+            curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'POST');
+            curl_setopt($ch, CURLOPT_POST, 1);
+            curl_setopt($ch, CURLOPT_POSTFIELDS, $payload);
+            curl_setopt($ch, CURLOPT_HTTPHEADER, [
+                'Content-type: ' . $contentType,
+                'Content-length: ' . mb_strlen($payload),
+            ]);
         }
 
-        throw new \RuntimeException('Post XML data set but curl PHP module is not installed or not enabled.');
+        // OTHERS
+        foreach ($this->curlOpts as $opt => $value) {
+            curl_setopt($ch, $opt, $value);
+        }
+        curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, 1);
+        curl_setopt($ch, CURLOPT_ENCODING, 'UTF-8');
+        curl_setopt($ch, CURLOPT_URL, $url);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+        curl_setopt($ch, CURLOPT_FOLLOWLOCATION, 1);
+        curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, $this->timeOut);
+
+        // EXECUTE and RESULT
+        $data     = curl_exec($ch);
+        $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+
+        // ANALYSE
+        if (false === $data) {
+            throw new \RuntimeException('Unhandled curl error: ' . curl_error($ch));
+        }
+
+        if (is_bool($data)) {
+            throw new \RuntimeException('Curl error: BOOL received, but STRING expected.');
+        }
+
+        if ($httpCode < 200 || $httpCode >= 300) {
```

Comment:
You mean the status code check? Not sure. I was just moving logically elements. The status check was already introduced on July, 30th 2020.

But it makes sense, since 400-codes are the good once ;-)